benchmark/single-source/FloatingPointParsing.swift

```diff
@@ -0,0 +1,177 @@
+//===--- FloatingPointParsing.swift -----------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// This test verifies the performance of parsing a floating point value
+// from a String.
+
+import TestsUtils
+
+public let FloatingPointParsing = [
+  BenchmarkInfo(
+    name: "FloatingPointParsing_Float_small",
+    runFunction: run_FloatingPointParsing_Float_small,
+    tags: [.validation, .api, .runtime, .String],
+    legacyFactor: 108),
```

Comment:
These are new benchmarks, so no need for legacy factors. Naming conventions for new benchmarks are [here](https://github.com/apple/swift/blob/master/benchmark/Naming.md).

Perhaps "ParseFloat.Float.small", "ParseFloat.Double.small", .... 

CC @palimondo